homeassistant/components/sensor/mysensors.py

```diff
@@ -11,145 +11,182 @@
-                node[child_id] = {}
-            for value_type, value in child.values.items():
-                if value_type not in node[child_id]:
-                    name = '{} {}.{}'.format(sensor.sketch_name, nid, child.id)
-                    node[child_id][value_type] = \
-                        MySensorsNodeValue(
-                            nid, child_id, name, value_type, is_metric, const)
-                    new_devices.append(node[child_id][value_type])
-                else:
-                    node[child_id][value_type].update_sensor(
-                        value, sensor.battery_level)
-
-        if new_devices:
-            _LOGGER.info("adding new devices: %s", new_devices)
-            add_devices(new_devices)
-
-    port = config.get(CONF_PORT)
-    if port is None:
-        _LOGGER.error("Missing required key 'port'")
-        return False
-
-    persistence = config.get(CONF_PERSISTENCE, True)
-    persistence_file = config.get(CONF_PERSISTENCE_FILE, 'mysensors.pickle')
-    version = config.get(CONF_VERSION, '1.4')
-
-    gateway = mysensors.SerialGateway(port, sensor_update,
-                                      persistence=persistence,
-                                      persistence_file=persistence_file,
-                                      protocol_version=version)
-    gateway.metric = is_metric
-    gateway.debug = config.get(CONF_DEBUG, False)
-    gateway.start()
-
-    if persistence:
-        for nid in gateway.sensors:
-            sensor_update('sensor_update', nid)
-
-    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP,
-                         lambda event: gateway.stop())
+    """Setup the mysensors platform for sensors."""
+    # Define the S_TYPES and V_TYPES that the platform should handle as states.
+    s_types = [
+        mysensors.CONST.Presentation.S_TEMP,
+        mysensors.CONST.Presentation.S_HUM,
+        mysensors.CONST.Presentation.S_BARO,
+        mysensors.CONST.Presentation.S_WIND,
+        mysensors.CONST.Presentation.S_RAIN,
+        mysensors.CONST.Presentation.S_UV,
+        mysensors.CONST.Presentation.S_WEIGHT,
+        mysensors.CONST.Presentation.S_POWER,
+        mysensors.CONST.Presentation.S_DISTANCE,
+        mysensors.CONST.Presentation.S_LIGHT_LEVEL,
+        mysensors.CONST.Presentation.S_IR,
+        mysensors.CONST.Presentation.S_WATER,
+        mysensors.CONST.Presentation.S_AIR_QUALITY,
+        mysensors.CONST.Presentation.S_CUSTOM,
+        mysensors.CONST.Presentation.S_DUST,
+        mysensors.CONST.Presentation.S_SCENE_CONTROLLER,
+        mysensors.CONST.Presentation.S_COLOR_SENSOR,
+        mysensors.CONST.Presentation.S_MULTIMETER,
+    ]
+    not_v_types = [
+        mysensors.CONST.SetReq.V_ARMED,
+        mysensors.CONST.SetReq.V_STATUS,
+        mysensors.CONST.SetReq.V_LIGHT,
+        mysensors.CONST.SetReq.V_LOCK_STATUS,
+    ]
+    v_types = []
+    for _, member in mysensors.CONST.SetReq.__members__.items():
```

Comment:
Since you only care about the members of this Enum, you can just iterate on `SetReq`. Also, since you want to end up with a list, consider this:
```python
v_types = [member for member in mysensors.CONST.SetReq
           if member.value not in not_v_types]
```